crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -2487,7 +2490,17 @@ impl<'db> TypeInferenceBuilder<'db> {
         self.infer_expression(expression)
     }
 
-    fn infer_arguments(&mut self, arguments: &ast::Arguments) -> CallArguments<'db> {
+    fn infer_arguments(
+        &mut self,
+        arguments: &ast::Arguments,
+        infer_as_type_expressions: bool,
```

Comment:
As mentioned elsewhere, we'll need per-argument granularity on this soon, but it doesn't have to be in this PR.

Reply:
Ok, I'll leave this as a follow-up task.